bundles/core/src/main/java/com/adobe/cq/wcm/core/components/services/link/LinkRequest.java

```diff
@@ -0,0 +1,59 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2021 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.services.link;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.osgi.annotation.versioning.ProviderType;
+
+/**
+ * Simple PoJo which contains required information to create a {@link com.adobe.cq.wcm.core.components.commons.link.Link} by the
+ * {@link LinkProcessor}
+ *
+ * @since com.adobe.cq.wcm.core.components.services.link 1.0.0
+ */
+@ProviderType
+public final class LinkRequest<T> {
```

Comment:
I'm not sure having a wrapper object here is that helpful, I find it a bit confusing, maybe we should pass the path, reference and attributes directly.

Reply:
I find the name confusing. If we decide to keep it, I'd rename it `LinkData`.